nexus/src/external_api/console_api.rs

```diff
@@ -177,11 +324,75 @@ pub async fn consume_credentials(
         )
         .await?;
 
-        match identity_provider {
-            IdentityProviderType::Saml(_saml_identity_provider) => {
-                todo!()
-            }
+        let (authenticated_subject, relay_state_string) =
+            match identity_provider {
+                IdentityProviderType::Saml(saml_identity_provider) => {
+                    let body_bytes = body_bytes.as_str()?;
+                    saml_identity_provider.authenticated_subject(
+                        &body_bytes,
+                        nexus.samael_max_issue_delay(),
+                    )?
+                }
+            };
+
+        let relay_state: Option<RelayState> =
+            if let Some(value) = relay_state_string {
+                Some(RelayState::from_encoded(value).map_err(|e| {
+                    HttpError::for_internal_error(format!("{}", e))
+                })?)
+            } else {
+                None
+            };
+
+        info!(
```

Comment:
nit: There are several info-level log messages in this function that I'd suggest replacing with `debug!` level instead.  My usual rule of thumb is that we can afford one "info"-level log message per request to the public API, and that's the one at the end of the request.

Some of these seem like they could use more context too so that someone looking at the log message understands where it came from or what it means.

Reply:
I don't think this is a nit, I wouldn't want to cause a flood of messages. Fixed in c7c2f172